### PR TITLE
shellcheck create_dummy_box.sh

### PR DESCRIPTION
**Pull request by konstruktoid**
Nitpicking and shellchecking create_dummy_box.sh

Signed-off-by: Thomas Sjögren <konstruktoid@users.noreply.github.com>

### CODE REVIEW

File: tests/tools/create_dummy_box.sh [hunk 1]
```diff
@@ -8,15 +8,15 @@ if [ $# -ne 1 ]; then
 	exit 1
 fi
 
-if [ -f "$DIR/dummy.box" ]; then
+if [ -f "${DIR}/dummy.box" ]; then
```
Comment (apatard) [inline tests/tools/create_dummy_box.sh:11]: Not related to this PR but looks like I produced a broken check. Should have been dummy-$provider.box not dummy.box.

File: tests/tools/create_dummy_box.sh [hunk 2]
```diff
@@ -8,15 +8,15 @@ if [ $# -ne 1 ]; then
...
-cd "$TMPDIR"
+TMPDIR="$(mktemp -d)"
+cd "${TMPDIR}" || exit 1
 echo "{ \"provider\": \"$PROVIDER\"}" > metadata.json
-tar czf "$DIR/dummy-$PROVIDER.box" .
-cd "$DIR"
-rm -rf "$TMPDIR"
+tar czf "${DIR}/dummy-$PROVIDER.box" .
+cd "${DIR}" || exit 1
+rm -rvf "${TMPDIR}"
```
Comment (apatard) [inline tests/tools/create_dummy_box.sh:22]: Do you have the reason of adding '-v' to rm ? I'm not against it. Only wondering.

File: tests/tools/create_dummy_box.sh [hunk 3]
```diff
@@ -8,15 +8,15 @@ if [ $# -ne 1 ]; then
...
+if [ -f "${DIR}/dummy.box" ]; then
 	echo "Box already created"
 	exit 0
 fi
 
 PROVIDER="$1"
-TMPDIR=`mktemp -d`
-cd "$TMPDIR"
+TMPDIR="$(mktemp -d)"
+cd "${TMPDIR}" || exit 1
```
Comment (apatard) [inline tests/tools/create_dummy_box.sh:18]: I'm not sure to understand the aim of adding '||exit 1'. We're running with 'set set -euxo pipefail` and the directory is created the line before. Same for the cd "${DIR}".

File: tests/tools/create_dummy_box.sh (hunk 1 above)
Comment (konstruktoid) [reply tests/tools/create_dummy_box.sh]: Updated.

File: tests/tools/create_dummy_box.sh (hunk 2 above)
Comment (konstruktoid) [reply tests/tools/create_dummy_box.sh:22]: `-v` just matches `set -x` so well

Comment (apatard) [reply tests/tools/create_dummy_box.sh:22]: oh ok

File: tests/tools/create_dummy_box.sh (hunk 3 above)
Comment (apatard) [reply tests/tools/create_dummy_box.sh:18]: ok. better be safe than sorry :)

**Review by apatard — APPROVED** [review]

**Pull request merged**